### PR TITLE
New: Add `no-useless-computed-key` rule

### LINKED ISSUE #5402: Proposing a "computed-property-as-needed" rule
With ES6 Computed Properties, one can do the following:

```javascript
let prop = 'b';
return {
  'a': 1,
  [prop]: 2,
  c: 3,
  ['d']: 4
}
```

Note how the brackets around the `['d']` property are superfluous. It could instead be written as `'d'` or `d`.

**When does this rule warn? Please describe and show example code:**

If enabled, the following would trigger:

```javascript
{
  ['x']: 1
}
```

Any property which does not need to be computed could trigger the rule.

**Is this rule preventing an error or is it stylistic?**

Stylistic

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

This rule affects basic ES6 syntax for situations where there are two equivalent ways to write the same code.

**Are you willing to create the rule yourself?**

I will attempt to if nobody else is interested.

### PR DESCRIPTION
**Pull request by BYK**
Fixes #5402.

### CODE REVIEW

**Comment by mysticatea** [comment]
Looks great to me.

File: docs/rules/no-useless-computed-key.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+The code can be rewritten as:
+
+```js
+var foo = {"a": "b"};
+```
+
+## Rule Details
+
+This rule aims to flag unnecessary usages of computed property keys.
```
Comment (scriptdaemon) [inline docs/rules/no-useless-computed-key.md]: To remain consistent with current doc updates, what do you think of this?

`This rule disallows unnecessary usage of computed property keys.`

File: lib/rules/no-useless-computed-key.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+var MESSAGE_UNNECESSARY_COMPUTED = "Unnecessarily computed property [{{property}}] found.";
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Disallow unnecessary computed property keys on objects",
```
Comment (pedrottimark) [inline lib/rules/no-useless-computed-key.js]: exactly as in README.md first letter of first word is lowercase

what do you think about s/on objects/in object literals/ here and in README.md and in rule doc .md file?

File: docs/rules/no-useless-computed-key.md (hunk 1 above)
Comment (BYK) [inline docs/rules/no-useless-computed-key.md]: Done, update coming.

File: lib/rules/no-useless-computed-key.js (hunk 2 above)
Comment (BYK) [inline lib/rules/no-useless-computed-key.js]: Okay to both.

**Comment by BYK** [comment]
@scriptdaemon @pedrottimark updated

**Comment by pedrottimark** [comment]
@byk Almost there, can you replace “on object literals” with “**in** object literals”
* in README.md
* in the rule .js file
* and also similarly update “on objects” the main heading of the rule .md file

**Comment by BYK** [comment]
@pedrottimark done

**Comment by pedrottimark** [comment]
Thank you, the docs LGTM

**Comment by ilyavolodin** [comment]
LGTM, thanks.

**Pull request merged**